exercises/practice/transpose/tests/transpose.cairo

```diff
@@ -0,0 +1,174 @@
+use transpose::transpose as transpose_matrix;
+
+#[test]
+fn empty_string() {
+    let mut input: Array<ByteArray> = array![];
+    let expected = array![];
+
+    let output = transpose_matrix(input);
+    assert_eq!(output, expected);
+}
+
+#[test]
+#[ignore]
+fn two_characters_in_a_row() {
+    let mut input: Array<ByteArray> = array!["A1"];
+    let expected = array!["A", "1"];
+
+    let output = transpose_matrix(input);
+    assert_eq!(output, expected);
+}
+
+#[test]
+#[ignore]
+fn two_characters_in_a_column() {
+    let mut input: Array<ByteArray> = array!["A", "1"];
+    let expected = array!["A1"];
+
+    let output = transpose_matrix(input);
+    assert_eq!(output, expected);
+}
+
+#[test]
+#[ignore]
+fn simple() {
+    let mut input: Array<ByteArray> = array!["ABC", "123"];
+    let expected = array!["A1", "B2", "C3"];
+
+    let output = transpose_matrix(input);
+    assert_eq!(output, expected);
+}
+
+#[test]
+#[ignore]
+fn single_line() {
+    let mut input: Array<ByteArray> = array!["Single line."];
+    let expected = array!["S", "i", "n", "g", "l", "e", " ", "l", "i", "n", "e", "."];
+
+    let output = transpose_matrix(input);
+    assert_eq!(output, expected);
+}
+
+#[test]
+#[ignore]
+fn first_line_longer_than_second_line() {
+    let mut input: Array<ByteArray> = array!["The fourth line.", "The fifth line."];
+    let expected = array![
+        "TT",
+        "hh",
+        "ee",
+        "  ",
+        "ff",
+        "oi",
+        "uf",
+        "rt",
+        "th",
+        "h ",
+        " l",
+        "li",
+        "in",
+        "ne",
+        "e.",
+        ". "
+    ];
+
+    let output = transpose_matrix(input);
+    assert_eq!(output, expected);
+}
+
+#[test]
+#[ignore]
+fn second_line_longer_than_first_line() {
+    let mut input: Array<ByteArray> = array!["The first line.", "The second line."];
+    let expected = array![
+        "TT",
+        "hh",
+        "ee",
+        "  ",
+        "fs",
+        "ie",
+        "rc",
+        "so",
+        "tn",
+        " d",
+        "l ",
+        "il",
+        "ni",
+        "en",
+        ".e",
+        " ."
+    ];
+
+    let output = transpose_matrix(input);
+    assert_eq!(output, expected);
+}
+
+#[test]
+#[ignore]
+fn mixed_line_length() {
+    let mut input: Array<ByteArray> = array![
+        "The longest line.", "A long line.", "A longer line.", "A line."
+    ];
+    let expected = array![
+        "TAAA",
+        "h   ",
+        "elll",
+        " ooi",
+        "lnnn",
+        "ogge",
+        "n e.",
+        "glr ",
```

Comment:
You should not pad to the right, i.e. add padding spaces to the rightmost columns (as per [the problem description](https://github.com/exercism/problem-specifications/blob/main/exercises/transpose/description.md?plain=1#L26)).

Copy/paste the actual expected output from [the canonical data](https://github.com/exercism/problem-specifications/blob/main/exercises/transpose/canonical-data.json#L156)